website/blog/2022-10-05-Harness-CI-Onboarding/harness-ci-onboarding.md

```diff
@@ -0,0 +1,259 @@
+
+### Prerequisites
+
+
+- Github Account ([Check out the other source code managers here](https://docs.harness.io/article/kqik8km5eb-source-code-manager-settings#source-code-manager-settings))
+- Docker Hub Account & Repository ([Other artifactory packages](https://docs.harness.io/article/66ykcm0sf0-build-and-push-to-gcr-step-settings))
+- Installation of K8s Delegate for Harness Delegate
+- About Harness CI
+- Features in Harness CI
+- Getting Started with your first pipeline
+- Troubleshooting frequently encountered issues
+- Conclusion: Developer Feedback on Harness CIE
+- What’s Next?
+
+This tutorial implements creation of a pipeline over a github repository thus you’ll be required to create a github account & host a project over a repository. To create a new repository on github follow these steps:
+
+1. Move to the upper-right corner on the github webapp & use the  drop-down menu to select New repository option.
+
+![create-new-repository](./repo-create.png)
+
+2. Type the name of the repository. The repository name is unique & cannot be same as already hosted repo names.
+
+![repo-name](./create-repository-desc.png)
+
+3. Select the repository visibility & click on create repository.
+
+![repo-public](./create-repository-public-private.png)
+
+![create-repo](./repo-create.png)
+
+### Installing Docker Engine
+
+To get started with Docker Engine make sure you meet the [prerequisites](https://docs.docker.com/engine/install/ubuntu/#prerequisites) then install Docker. Older versions of Docker were called ```docker```, ```docker.io```, or ```docker-engine```. If these are installed uninstall them with this command.
+
+```bash
+sudo apt-get remove docker docker-engine docker.io containerd runc
+```
+
+Before installing docker engine we’ll first setup the repository & then update docker from the repository. To set-up the repository use the following command:
+
+Before installing docker engine we’ll first setup the repository & then update docker from the repository. To set-up the repository use the following command:
+
+
+```bash
+$ sudo apt-get update
+
+$ sudo apt-get install \
+    ca-certificates \
+    curl \
+    gnupg \
+    lsb-release
+```
+
+We’ll then add the GPG key and setup the repository using the following command.
+
+```bash
+$ sudo mkdir -p /etc/apt/keyrings
+$ curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo gpg --dearmor -o /etc/apt/keyrings/docker
+```
+
+```bash
+$ echo \
+  "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/docker.gpg] https://download.docker.com/linux/ubuntu \
+  $(lsb_release -cs) stable" | sudo tee /etc/apt/sources.list.d/docker.list > /dev/null
+```
+
+Now to install Docker Engine we’ll update the apt package and install the latest version, containered & Docker compose.
+
+```bash
+$ sudo apt-get update
+$ sudo apt-get install docker-ce docker-ce-cli containerd.io docker-compose-plugin
+```
+
```

Comment:
If it's already present in the docker site, then let's refer to the link as they'll update the site often!